app/helpers/loss_report_helper.rb

```diff
@@ -0,0 +1,189 @@
+module LossReportHelper
+
+  def cmr_client
+    Cmr::Client.client_for_environment(Rails.configuration.cmr_env, Rails.configuration.services)
+  end
+
+  def token
+    if session[:login_method] == 'launchpad'
+      session[:launchpad_cookie]
+    elsif session[:login_method] == 'urs'
+      session[:access_token]
+    end
+  end
+
+  def loss_report_output(hide_items=true, disp='text')
+    # depending on the input selection (json or text) a comparison string/hash is created and displayed in-browser
+
+    # @collection_error is true if there is an error in the translation that is performed by prepare_collections in the collections_controller
+    if !@collection_error
+      orig_xml,conv_xml = @original_collection_native_xml, @translated_collection_native_xml
+      orig_h,conv_h  = @original_collection_native_hash, @translated_collection_native_hash
+    else
+      return 'Failure to get_concept or translate_collection' if disp == 'text'
+      return {"error"=>"Failure to get_concept or translate_collection"} if disp == 'json'
+    end
+
+    # ISO and DIF collections (in XML form) contain namespaces that cause errors in the below comparison.
+    # Specifically, when nodes are evaluated individually, (their namespace definitions remaining at the top of the xml)
+    # their prefixes are undefined in the scope of the evaluation and therefore raise errors. Removing the namespaces
+    # eliminates this issue.
+    if @content_type.include?('iso') || @content_type.include?('dif')
+      orig = Nokogiri::XML(orig_xml) { |config| config.strict.noblanks } .remove_namespaces!
+      conv = Nokogiri::XML(conv_xml) { |config| config.strict.noblanks } .remove_namespaces!
+    else
+      orig = Nokogiri::XML(orig_xml) { |config| config.strict.noblanks }
+      conv = Nokogiri::XML(conv_xml) { |config| config.strict.noblanks }
+    end
+
+    arr_paths = Array.new # This array is used to keep track of the paths that lead to arrays that have already been mapped
+    text_output = String.new if disp == 'text'
```

Comment:
1. Can you add a comment here that the text and json outputs seem to be a good candidate for using dependency injection?
2. I think since lines 40-44 are all checking what `disp` is they can be combined.
3. It seems that whatever `disp` is not, that variable will be `nil`, and we will be passing that around the other methods, right? If that's the case I think it would be clearer to just declare it as `nil`
4. This is super minor and you don't have to change it, but I think `display` would be a bit more clear than `disp` and not a lot more characters

Reply:
Just made changes 1,2, and 4. I'm not entirely sure what you mean for #3. If you could expand a little more, that'd be great.